provisioners/bash/amazon-linux.sh

```diff
@@ -2,6 +2,13 @@
 set -o nounset
 set -o errexit
 
-# Placeholder for anything Amazon Linux specific shell provisioning
+yum -y upgrade
+rpm -ivh --force https://yum.puppetlabs.com/puppetlabs-release-pc1-el-6.noarch.rpm
+yum -y install puppet-agent
+
+# when installing puppet5, we got this error as Amazon Linux does not have systemd
```

Comment:
Could you please try Amazon Linux 2 (201712) https://aws.amazon.com/amazon-linux-2/release-notes/ ?
That version is the first one with systemd, so that should solve this error and allows you to install puppet5, and hopefully there's no need to downgrade the gems. Also interested to see if the device names would work with default setting.